controller/node_controller.go

```diff
@@ -837,6 +837,9 @@ func (nc *NodeController) updateReadyDiskStatusReadyCondition(node *longhorn.Nod
 				longhorn.DiskConditionTypeReady, longhorn.ConditionStatusTrue,
 				"", fmt.Sprintf("Disk %v(%v) on node %v is ready", diskName, diskInfoMap[diskName].Path, node.Name),
 				nc.eventRecorder, node, corev1.EventTypeNormal)
+			if diskStatus.Type == longhorn.DiskTypeFilesystem {
+				diskStatus.Type = longhorn.DiskType(info.DiskStat.Type)
+			}
```

Comment:
I do not think it is appropriate to set this here (since this function is for updating the ready condition. We should probably uplevel it to `syncDiskStatus` (or a new function within `syncDiskStatus`.

I don't think I like changing the meaning of the `diskStatus.Type` field, which differentiates between v1 and v2 disks. I prefer a new field like `diskStatus.filesystemType` (which can be ignored for v2 disks).

Reply:
I agree to add a new field to keep the file system type as well.
Added a new field `FSType` in `DiskStatus` to record the file system type.
Thanks.

Reply:
Also added a new function to update the `FSType` in `syncDiskStatus`